src/app/state/account/saga.ts

```diff
@@ -126,13 +149,14 @@ export function* fetchingOnAccountPage() {
           }
 
           const staleBalances = yield* select(selectAccount)
+          const hasPendingTxs = yield* select(hasAccountUnknownPendingTransactions)
           if (
             staleBalances.available !== refreshedAccount.available ||
             staleBalances.delegations !== refreshedAccount.delegations ||
-            staleBalances.debonding !== refreshedAccount.debonding
+            staleBalances.debonding !== refreshedAccount.debonding ||
+            hasPendingTxs
           ) {
             // Wait for oasisscan to update transactions (it updates balances faster)
-            yield* delay(TRANSACTIONS_UPDATE_DELAY)
             yield* call(fetchAccount, startAction)
```

Comment:
Did you try if this works?

Reply:
Yes, there is no need for delay anymore because of `hasPendingTxs`. Now we know when account is out of sync with transaction history, and keep making requests until nonce will match.

Reply:
Unless you can think of a case, where that would not work?

Reply:
ah, it was my edge case.
I made transfer to myself, bypassing "Cannot send to yourself" error:
- (on chain: made a transaction, incremented nonce, didn't change balance)
- frontend adds transaction to selectPendingTransactionForAccount
- account nonce isn't updated so hasAccountUnknownPendingTransactions is false
- ACCOUNT_REFETCHING_INTERVAL: fetches account, but detects no balance change and no pending txs
- ACCOUNT_REFETCHING_INTERVAL: fetches account, but detects no balance change and no pending txs

Transfer to another account:
- (on chain: made a transaction, incremented nonce, changed balance)
- frontend adds transaction to selectPendingTransactionForAccount
- account nonce isn't updated so hasAccountUnknownPendingTransactions is false
- ACCOUNT_REFETCHING_INTERVAL: fetches account, **detects balance change**:
  - saves new account info into state (updated balance and **updated nonce**)
  - fetches and saves transactions but they aren't changed
- ACCOUNT_REFETCHING_INTERVAL: fetches account, no balance change, **has pending txs** (updated nonce)
  - fetches and **saves transactions**